### ISSUE TITLE
`serverpod_auth_google_flutter`: `PlatformException: Failed to disconnect` when calling `googleSignIn.disconnect()`

### ISSUE THREAD

**Issue by lukehutch** [serverpod/serverpod#735]
**Describe the bug**

I am using the git latest version of `serverpod_auth` as a Flutter dependency. When I try signing in using Google, I get:

```
I/flutter (31831): serverpod_auth_google: GoogleSignIn
W/Parcel  (31831): Expecting binder but got null!
I/flutter (31831): serverpod_auth_google: Signing out from google
I/flutter (31831): serverpod_auth_google: PlatformException(status, Failed to disconnect., null, null)
I/flutter (31831): #0      StandardMethodCodec.decodeEnvelope
message_codecs.dart:653
I/flutter (31831): #1      MethodChannel._invokeMethod
platform_channel.dart:315
I/flutter (31831): <asynchronous suspension>
I/flutter (31831): #2      MethodChannel.invokeMapMethod
platform_channel.dart:518
I/flutter (31831): <asynchronous suspension>
I/flutter (31831): #3      GoogleSignIn._callMethod
google_sign_in.dart:274
I/flutter (31831): <asynchronous suspension>
```

The line that causes the issue is marked `// (*)` below (`auth.dart:101`):

```dart
    // Store the user info in the session manager.
    var sessionManager = await SessionManager.instance;

    await sessionManager.registerSignedInUser(
      serverResponse.userInfo!,
      serverResponse.keyId!,
      serverResponse.key!,
    );

    // Authentication with server is complete, we can sign out from Google locally
    if (kDebugMode) print('serverpod_auth_google: Signing out from google');
    await googleSignIn.signOut();

    if (kDebugMode) await googleSignIn.disconnect();  // (*)
```

**To Reproduce**

Create a new project that uses `serverpod_auth`, but `git clone` the `serverpod` repo, and depend upon the local copy of the `serverpod_auth` libraries, not the pub versions of the libraries.

**Comment by vlidholt** [comment]
Are you using the same version for both Serverpod and the auth module?

**Comment by lukehutch** [comment]
Yes, I'm depending on git versions of everything

**Comment by yahu1031** [comment]
Did you check that with the normal package?

**Comment by lukehutch** [comment]
@yahu1031 I just went back and checked with serverpod 1.0.1, and the error occurs with that version too. So in fact this probably started happening due to an update of Flutter (now version 3.7.7), not serverpod. Flutter seems to now be more aggressive at unmounting the context after the widget tree is no longer being built.

> Edit: AFAIR first disconnect then sign out. Did you try this?

The `serverpod_auth` code already does that. In `serverpod_auth_google_flutter`, file `auth.dart`, there is the following code:

```dart
    // Authentication with server is complete, we can sign out from Google locally
    if (kDebugMode) print('serverpod_auth_google: Signing out from google');
    await googleSignIn.signOut();

    if (kDebugMode) await googleSignIn.disconnect();
```

The `googleSignIn.disconnect();` line is the one that fails.

At a very minimum, this should be as follows:

```dart
    try {
      if (kDebugMode) await googleSignIn.disconnect();
    } catch (e) { print('Exception while disconnecting from Google: $e'); }
```

**Comment by yahu1031** [comment]
@lukehutch I hope this ticket can be closed as it was resolved in #753
